src/components/navigation-bar/navigation-bar.test.js

```diff
@@ -0,0 +1,169 @@
+import React from "react";
+import Box from "../box/box.component";
+import NavigationBar from "./navigation-bar.component";
+import { Menu, MenuItem } from "../menu";
+import CypressMountWithProviders from "../../../cypress/support/component-helper/cypress-mount";
+
+import navigationBar from "../../../cypress/locators/navigation-bar";
+
+const specialCharacters = ["mp150ú¿¡üßä", "!@#$%^*()_+-=~[];:.,?{}&\"'<>"];
+const testData = "cypressData";
+const variants = [
+  ["black", "rgb(0, 0, 0)"],
+  ["light", "rgb(230, 235, 237)"],
+  ["white", "rgb(255, 255, 255)"],
+  ["dark", "rgb(0, 50, 76)"],
+];
+
+const NavigationBarComponent = ({ children, ...props }) => {
+  const isChildren = children != null ? children : "Cypress tests";
+  return <NavigationBar {...props}>{isChildren}</NavigationBar>;
+};
+
+context("Testing NavigationBar component", () => {
+  describe("should render NavigationBar component", () => {
+    it.each(specialCharacters)(
+      "should render NavigationBar component with %s as a children",
+      (childrenValue) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent>{childrenValue}</NavigationBarComponent>
+        );
+
+        navigationBar().should("have.text", childrenValue);
+      }
+    );
+
+    it("should render NavigationBar component with ariaLabel prop", () => {
+      CypressMountWithProviders(
+        <NavigationBarComponent ariaLabel={testData} />
+      );
+
+      navigationBar().should("have.attr", "aria-label", testData);
+    });
+
+    it.each(variants)(
+      "should render NavigationBar component with %s as a navigationType",
+      (navigationType, color) => {
+        CypressMountWithProviders(
+          <NavigationBarComponent navigationType={navigationType} />
+        );
+
+        navigationBar().should("have.css", "background-color", color);
+      }
+    );
+
+    it.each([
+      [true, "not.exist"],
+      [false, "be.visible"],
```

Comment:
Hi @robinzigmond, you are right. But, cypress cares if the element is visible or not (but it definitely should be rendered in DOM).

Reply:
I think it would be good to verify the same property for both scenarios.  I would think visibility would be the one to check. But you could also check both visibility and existence for both scenarios?